src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -695,8 +700,12 @@ impl<'cfg> RustcTargetData<'cfg> {
         if requested_kinds.iter().any(CompileKind::is_host) {
             let ct = CompileTarget::new(&rustc.host)?;
             target_info.insert(ct, host_info.clone());
-            target_config.insert(ct, host_config.clone());
-        }
+            if target_applies_to_host {
+                target_config.insert(ct, host_config.clone());
+            } else {
+                target_config.insert(ct, config.target_cfg_triple(&rustc.host)?);
```

Comment:
I'm not sure I understand this line, if `target_applies_to_host` is `false`, then why is the target configuration loaded here?

(I was under the impresion this line doesn't need to change?)

Reply:
Yeah, it's because we're only trying to load the target config here...if you look above host_config is the same as the target config if `target_applies_to_host` is true but not if it is false...which is why we clone when true...but if it's false then we have to load it separately.